homework-week3.js

```diff
@@ -0,0 +1,196 @@
+favoriteAnimals.indexOf("meerkat");     // we find the index of the element ---- it is : 1
+console.log("The item you are looking for is at index: " + favoriteAnimals.indexOf("meerkat"));
+//removed = favoriteAnimals.splice(1, 1);  // we delete it using .splice(the index, how many times)
+//console.log(favoriteAnimals);
+
+
+// ------ More JavaScript --------
+
+// 3 arguments function that returns the sum
+function sum (a,b,c) {
+    return a+b+c;
+}
+sum (1,2,3);   // 6
+
+//---------colorCar function -----------
+function colorCar (color) {
+    return "a "+ color + " car";   //the input should be a string .... I tried a lot but couldn't use red as a color not as a string????
+} 
+colorCar("red");   //"a red car"
+
+//--------vehicle function (obj)---------
+const personalInfo = {
+    name: "Ghufran", 
+    age: 34,
+};
+function myFunction (obj){      
+    return  console.log(personalInfo.name, personalInfo.age);
+}
+myFunction(personalInfo) // Ghufran 34
+
+
+// ------ vehicleType function-----------
+function vehicleType (color,code) {
+    if (code === 1){
+        return "a " + color + " car";
+    } else {                          
+        return "a " + color + " motorbike";
+    } 
+}
+vehicleType("blue", 2)    //"a blue motorbike"
+
+/*
+if (3 === 3) {
+  console.log("yes");
+} else {
+  console.log("no");
+}
+*/
+const message = (3 === 3) ? "yes" : "no";
+console.log(message);  //      yes
+
+//--------vehicle function (color, code, age)---------
+function vehicle (color, code, age){
+    if(age > 1 && code === 1){
+        return "a " + color + " used" + " car";
+    } else if(age > 1 && code === 2){
+        return "a " + color + " used" + " motorbike";
+    }
+    else if(age = 1 && code === 1){
+        return "a " + color + " new" + " car";
+    }
+    else if (age =1 && code === 2){
+        return "a " + color + " new" + " motorbike";
+    }
+}
+
+
+
+//---------a list of vehicles--------- 
+const listOfVehicles = ["motorbike", "car", "caravan", "bike"];
+//----------get the third element---------
+listOfVehicles[2];   //"caravan"
+
+//--------vehicle function (color,item , age)---------
+function vehicle (color,item , age){
+    if (age >= 5 && item < listOfVehicles.length) {
+        return "a " + color + " used " + listOfVehicles[item];
+    } 
+    else if (age < 5 && item < listOfVehicles.length){
+        return "a " + color + " new " + listOfVehicles[item];
+    } else {
+        return "Item is not defined";
+    }
+  }
+
+vehicle("green", 3, 1)    //"a green new bike"
+
+// -------plurals/ commas + space---------
+//const listOfVehicles = ["motorbike", "car", "caravan", "bike"];     (mentioned above)
+function loopFunction(listOfVehicles){
+    for (let i=0; i < listOfVehicles.length; i++){
+       listOfVehicles[i] = listOfVehicles[i] + "s";
+    }
+    clearListOfVehicles = listOfVehicles.join(', ');   // I couldn't replace the (,) with (and), how?!!!!
```

Comment:
It takes a bit more work, you need to count the items in the loop and check if it's the last item, if it is the last item put `and` in front of it.

Reply:
I know it is toooooo late, but I could not sole the 'and ' problem

Reply:
You are inside the for loop so you have access to the `i` which holds the index, for every loop you know the `index`, next you want to know the length of the array, which you also know `listOfVehicles.length`. 

The next thing to do is check if the `index` is the same as the length of the array minus one. If that is the case join with 'and' otherwise join with ','.